cwl/src/main/scala/cwl/MyriadOutputTypeToWomType.scala

```diff
@@ -17,10 +14,17 @@ object MyriadOutputTypeToWomType extends Poly1{
     _.fold(MyriadOutputInnerTypeToWomType)
   }
 
-  implicit def acwl: Aux[Array[MyriadOutputInnerType], WomType] = at[Array[MyriadOutputInnerType]] {
-    _.toList.foldMap(i => Set(i)).map(_.fold(MyriadOutputInnerTypeToWomType)).toList match {
-      case head :: Nil => WomArrayType(head)
-      case _ => throw new RuntimeException("Wom does not provide an array of >1 types")
+  implicit def acwl: Aux[Array[MyriadOutputInnerType], WomType] = at[Array[MyriadOutputInnerType]] { types =>
+    types.partition(_.select[CwlType].contains(CwlType.Null)) match {
+      // If there's a single non null type, use that
+      case (Array(), Array(singleNonNullType)) =>
+        singleNonNullType.fold(MyriadOutputInnerTypeToWomType)
+      // If there's a null type and a single non null type, it's a WomOptionalType
+      case (Array(_), Array(singleNonNullType)) =>
+        WomOptionalType(singleNonNullType.fold(MyriadOutputInnerTypeToWomType))
+      // Leave other "Coproduct types" unsupported for now
+      case _ =>
+        throw new NotImplementedError("Multi types not supported yet")
```

Comment:
DRY this up with `MyriadInputTypeToWomType`

Reply:
😢  Although the logic is the same, almost every type and object is different.